tables/schema.py

```diff
@@ -0,0 +1,281 @@
+from collections import defaultdict
```

Comment:
What is a table schema? How is it used etc? Could this go in a module docstring here?

Reply:
Added some guidance to the module docstring with an example